### ISSUE TITLE
Closing NAPALM connections automatically

### ISSUE THREAD

**Issue by respawner** [nornir-automation/nornir#161]
When using NAPALM plugins connections stay open and can cause scripts to hang while waiting for the connections to timeout.

Forcing the connections to close using a task helps with this.

```python
task.host.connections["napalm"].close()
```

Maybe this could be done automatically? It'll require to find a way to know which connections are not useful anymore.

**Comment by ktbyers** [comment]
How would we know that you didn't want to use the connection any more (i.e. in a subsequent task)?

The process is probably to have the code writer to explicitly close the connection in some way.

**Comment by respawner** [comment]
Yep that's complicated. I opened the issue because @dbarrosop wanted to track it.

I don't see any simple way to do this. Maybe proposing a task closing all NAPALM connections could be an answer to this.

**Comment by dbarrosop** [comment]
I thought we could look into creating a “close” task so it’s easier for the user to explicitly close the connections or maybe do it automatically via the garbage collector by implementing some code in the __del__ magic method.

**Comment by ogenstad** [comment]
I think the close task, or perhaps something like a context manager for Nornir could make sense. There might be scenarios where users don't want the connections to be terminated. (I've seen Huawei switches where a local script running on them stops running if the ssh connection dies).

We could have some task which just iterates all of the hosts' connection dictionaries and calls a disconnect function. The main issue is perhaps that all of the connection plugins be consistent and support the same close method.

**Comment by dbarrosop** [comment]
For reference, some work that will enable this is being done in #189

**Comment by dmfigol** [comment]
@dbarrosop 
I think this is fixed with #195 and could be closed?
